tests/storages_tests/test_storages.py

```diff
@@ -352,336 +452,406 @@ def test_get_trial_number_from_id(storage_mode):
 
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_set_trial_user_attr(storage_mode: str) -> None:
 
-@parametrize_storage
-def test_set_and_get_trial_system_attr(storage_init_func):
-    # type: (Callable[[], BaseStorage]) -> None
+    with StorageSupplier(storage_mode) as storage:
+        trial_id_1 = storage.create_new_trial(storage.create_new_study())
 
-    storage = storage_init_func()
-    study_id = storage.create_new_study()
-    trial_id_1 = storage.create_new_trial(study_id)
+        def check_set_and_get(trial_id, key, value):
+            # type: (int, str, Any) -> None
 
-    def check_set_and_get(trial_id, key, value):
-        # type: (int, str, Any) -> None
+            storage.set_trial_user_attr(trial_id, key, value)
+            assert storage.get_trial(trial_id).user_attrs[key] == value
 
-        storage.set_trial_system_attr(trial_id, key, value)
-        assert storage.get_trial_system_attrs(trial_id)[key] == value
+        # Test setting value.
+        for key, value in EXAMPLE_ATTRS.items():
+            check_set_and_get(trial_id_1, key, value)
+        assert storage.get_trial(trial_id_1).user_attrs == EXAMPLE_ATTRS
 
-    # Test setting value.
-    for key, value in EXAMPLE_ATTRS.items():
-        check_set_and_get(trial_id_1, key, value)
-    system_attrs = storage.get_trial(trial_id_1).system_attrs
-    assert system_attrs == EXAMPLE_ATTRS
+        # Test overwriting value.
+        check_set_and_get(trial_id_1, "dataset", "ImageNet")
 
-    # Test overwriting value.
-    check_set_and_get(trial_id_1, "dataset", "ImageNet")
+        # Test another trial.
+        trial_id_2 = storage.create_new_trial(storage.create_new_study())
+        check_set_and_get(trial_id_2, "baseline_score", 0.001)
+        assert len(storage.get_trial(trial_id_2).user_attrs) == 1
+        assert storage.get_trial(trial_id_2).user_attrs["baseline_score"] == 0.001
 
-    # Test another trial.
-    trial_id_2 = storage.create_new_trial(study_id)
-    check_set_and_get(trial_id_2, "baseline_score", 0.001)
-    system_attrs = storage.get_trial(trial_id_2).system_attrs
-    assert system_attrs == {"baseline_score": 0.001}
+        # Cannot set attributes of non-existent trials.
+        non_existent_trial_id = max({trial_id_1, trial_id_2}) + 1
+        with pytest.raises(KeyError):
+            storage.set_trial_user_attr(non_existent_trial_id, "key", "value")
 
+        # Cannot set attributes of finished trials.
+        storage.set_trial_state(trial_id_1, TrialState.COMPLETE)
+        with pytest.raises(RuntimeError):
+            storage.set_trial_user_attr(trial_id_1, "key", "value")
 
-@parametrize_storage
-def test_get_all_study_summaries(storage_init_func):
-    # type: (Callable[[], BaseStorage]) -> None
 
-    storage = storage_init_func()
-    study_id = storage.create_new_study()
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_system_user_attrs(storage_mode: str) -> None:
```

Comment:
I guess this function name is actually `test_get_trial_system_attrs`.
```suggestion
def test_get_trial_system_attrs(storage_mode: str) -> None:
```